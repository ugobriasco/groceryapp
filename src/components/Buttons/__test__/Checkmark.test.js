import React from 'react';
import renderer from 'react-test-renderer';

import {Checkmark, styles} from '../';
import buildStyles from '../../../config/styles';

//do it if you care it
beforeAll(() => {
	buildStyles();
})

it('renders correctly', () => {
	const tree = renderer.create(<Checkmark />)

});

it ('styles an object', () => {
	expect(typeof styles).toBe('object');
});

it('renders a custom backgroundColor passed via props', () => {
	const rendered = renderer.create(<Checkmark iconBackground='red'/>);
	expect(rendered).toMatchSnapshot();
});

it('renders the defined background if isVisible is passed via props', () => {
	const rendered = renderer.create(<Checkmark isVisible={true}/>);
	expect(rendered).toMatchSnapshot();
});

it('renders a void component if checkmark passed via props is false', () => {
	const rendered = renderer.create(<Checkmark checkmark={false}/>);
	expect(rendered).toMatchSnapshot();
});