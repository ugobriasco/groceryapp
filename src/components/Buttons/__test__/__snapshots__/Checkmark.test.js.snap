// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a custom backgroundColor passed via props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "width": 30,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "width": 18,
      }
    }
  />
</View>
`;
